index_documents.go

```diff
@@ -70,80 +39,30 @@ func (i Index) addDocuments(documentsPtr interface{}, contentType string, primar
 		i.PrimaryKey = primaryKey[0] //nolint:golint,staticcheck
 		endpoint = "/indexes/" + i.UID + "/documents?primaryKey=" + primaryKey[0]
 	}
+
+	function := ""
+	if httpMethod == http.MethodPost {
+		function = "Add"
+	} else if httpMethod == http.MethodPut {
+		function = "Update"
+	}
+
 	req := internalRequest{
 		endpoint:            endpoint,
-		method:              http.MethodPost,
+		method:              httpMethod,
 		contentType:         contentType,
 		withRequest:         documentsPtr,
 		withResponse:        resp,
 		acceptedStatusCodes: []int{http.StatusAccepted},
-		functionName:        "AddDocuments",
+		functionName:        function + "Documents",
 	}
 	if err = i.client.executeRequest(req); err != nil {
 		return nil, err
 	}
 	return resp, nil
 }
```

Comment:
This method isn't needed you can call `addDocuments()` or `updateDocuments()` instead